crates/primitives/src/botanix/mint_validation.rs

```diff
@@ -197,23 +200,56 @@ pub fn try_parse_mint_event(
         let mut proofs = Vec::new();
         let mut offset = 0;
         while offset < meta_bytes.len() {
-            let (proof, proof_size) =
-                PeginMeta::deserialize(&meta_bytes[offset..]).map_err(|e| {
-                    let err = ParseMintEventError::InvalidPeginData {
-                        error: e,
-                        revert_address: destination,
-                        revert_amount: amount,
-                    };
-                    error!("Failed to parse pegin meta: {:?}", err);
-                    err
-                })?;
+            let version = deserialize_meta_version(&meta_bytes.clone()[offset..]).map_err(|e| {
+                let err = ParseMintEventError::InvalidPeginData {
+                    error: e,
+                    revert_address: destination,
+                    revert_amount: amount,
+                };
+                error!("Failed to parse pegin meta version: {:?}", err);
+                err
+            })?;
+            let (proof, proof_size) = match version {
+                PEGIN_META_VERSION_V0 => {
+                    let (meta, size) = PeginMeta::deserialize(&meta_bytes[offset..]).map_err(|e| {
+                        let err = ParseMintEventError::InvalidPeginData {
+                            error: e,
+                            revert_address: destination,
+                            revert_amount: amount,
+                        };
+                        error!("Failed to parse pegin meta: {:?}", err);
```

Comment:
```suggestion
                        error!("Failed to parse pegin v0 meta: {:?}", err);
```
And same change for the one below